Template/Tag/MatomoTag.web.js

```diff
@@ -117,38 +117,43 @@
                     }
                     configuredTrackers[variableName] = tracker;
 
-                    if (matomoConfig.disableCookies) {
-                        tracker.disableCookies();
+                    if (matomoConfig.requireCookieConsent) {
+                      	tracker.requireCookieConsent();
                     }
 
-                    if (matomoConfig.requireCookieConsent) {
-                        tracker.requireCookieConsent();
+                    if (matomoConfig.disableBrowserFeatureDetection && typeof tracker.disableBrowserFeatureDetection === 'function') {
+                        tracker.disableBrowserFeatureDetection();
                     }
 
-                    if (matomoConfig.requireConsent) {
-                        tracker.requireConsent();
+                    if (matomoConfig.disableCookies) {
+                        tracker.disableCookies();
                     }
 
                     if (matomoConfig.enableCrossDomainLinking) {
                         tracker.enableCrossDomainLinking();
                     }
 
-                    if (matomoConfig.setSecureCookie) {
-                        tracker.setSecureCookie(true);
+                    if (matomoConfig.requireConsent) {
```

Comment:
@AltamashShaikh assuming we are now calling all the methods in the same order as in the tracker file, we would maybe need to call `requireConsent` last? In https://github.com/matomo-org/matomo/blob/4.8.0/js/piwik.js#L7126 it shows up after the cookie domain

Reply:
Moved, now calling before `enableDoNotTrack` and after `enableLinkTracking`